.github/workflows/update-rock.yaml

```diff
@@ -56,7 +56,8 @@ jobs:
         shell: bash
         run: |
           release=$(yq '.parts.${{ inputs.rock-name }}["source-tag"]' $GITHUB_WORKSPACE/main/rockcraft.yaml)
-          if [ "${release}" != "${{steps.latest-release.outputs.release}}" ]; then
+          if [ "${release}" != "${{steps.latest-release.outputs.release}}" ] && \
+            [[ "${release}" > "${{steps.latest-release.outputs.release}}" ]]; then
```

Comment:
Afaik there is no built-in [semver comparison in bash](https://stackoverflow.com/questions/4023830/how-to-compare-two-strings-in-dot-separated-version-format-in-bash).
Am I missing something?

Reply:
No, there's not. `sort` can do it with `-V`, but string comparisons are better than they have any right to be:

```bash
#!/usr/bin/env bash

[[ "1.2.3" > "1.2.0" ]] && echo "Yes" || echo "No"
[[ "3.2.3" > "1.0.0" ]] && echo "Yes" || echo "No"
[[ "2.2.3" > "1.20.0" ]] && echo "Yes" || echo "No"
[[ "1.2.3" > "1.2.0" ]] && echo "Yes" || echo "No"
[[ "v1.2.3" > "v1.2.0" ]] && echo "Yes" || echo "No"
[[ "v3.2.3" > "v1.0.0" ]] && echo "Yes" || echo "No"
[[ "v2.2.3" > "v1.20.0" ]] && echo "Yes" || echo "No"
[[ "v1.2.3" > "v1.2.0" ]] && echo "Yes" || echo "No"
```

Reply:
Wouldn't string comparison say `"1.2" > "1.10`? This does not seem like a safe way to compare versions.